text/0047-switch-to-advanced-docking-system.md

```diff
@@ -0,0 +1,302 @@
+
+- QWidget based (WIP CSS)
+
+[![QWidget titlebar](./0047-switch-to-advanced-docking-system/linux_qwidget_floating_dock_titlebar.png)]()
+
+The QWidget one is used by default when using KWin based Desktop Environement like Plasma. And this titlebar requires some CSS in the themes to match it.
+
+So the QWidget based titlebar usage will be enforced thanks to a flag for the Dock Manager to "unify" experience between desktop environement.
+Theme makers may ignore to theme this bar.
+
+### Dock state
+*The per profile dock state feature is not taken into account to describe those changes.*
+
+In the global config (`global.ini`).
+- `"dockState"` is kept for for backward compatiblity and no longer overwitten. It will be used if the following state is not present.
+- `"windowState"` is the state of the main window, since it does not store only the state of legacy docks.
+- `"advDockState"` is the state of the dock manager which contain only the states of any ADS dock.
+
+Service integrations only save `"advDockProfile"`
+
+ADS dock state is compressed by default but behind the scene it's XML.
+
+### OBSAdvDock
+OBSAdvDock is a class which inherit `ads::CDockWidget` class.
+
+This class has a constructor which require a QWidget and setup some things arround the widget and set some connections.
+
+It adds a warning message when closing a dock from a close dock button.
+
+It allows to reset a dock position after a UI reset or a layout change with a possibility to reset the size if set beforehand.
+
+### Custom Browser docks and BrowserAdvDock
+The custom browser docks feature is modified to use BrowserAdvDock which inherit OBSAdvDock.
+
+Those docks are stored in the dock manager and their names is stored in a QStringList for browser docks to be able to get the dock from the manager to modify it like changing the URL.
+
+Each of those browser dock is named `extraBrowser_$UUID` where `$UUID` is replaced by the dock UUID to have a really unique name.
+
+### Service integration docks
+
+Like custom browser docks, those integration are modified to use BrowserAdvDock. But their names is stored in the QStringList for plugins extra docks.
+
+Those docks are named `obs-$SERVICE_$DOCK_NAME` where `$SERVICE` is the service name in lowercase and `$DOCK_NAME` the name of the dock.
+
+`"dockState"` will be imported from the integration config if `"windowState"` is not present.
+
+### Reset UI action
+1. Legacy dock are hidden
+2. The default state written in XML is applied
+3. Each not shown OBSAdvDock/BrowserAdvDock dock have its position and size reseted.
+
+## Legacy dock
+
+**TODO: Talk about `obs_frontend_add_custom_qdock()`**
+
+The frontend API method `obs_frontend_add_dock()` is put in deprecation. And if dual switch only Qt5 and on Qt 6 the method is removed or does nothing, this is possible because plugin will need to be rebuilt agaisnt Qt6.
```

Comment:
This needs updating based on current situation.

Reply:
This is why I added TODOs, I'm working on it.